src/org/exist/storage/BrokerPools.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * eXist Open Source Native XML Database
+ * Copyright (C) 2003-2016 The eXist-db Project
+ * http://exist-db.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.exist.storage;
+
+import net.jcip.annotations.GuardedBy;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.exist.EXistException;
+import org.exist.util.Configuration;
+import org.exist.util.DatabaseConfigurationException;
+import org.exist.util.function.ConsumerE;
+
+import java.util.*;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * This abstract class really just contains the static
+ * methods for {@link BrokerPool} to help us organise the
+ * code into smaller understandable chunks and reduce the
+ * complexity when understanding the concurrency
+ * constraints between one and many BrokerPools
+ *
+ * @author Adam Retter <adam@exist-db.org>
+ * @author Wolfgang Meier <wolfgang@exist-db.org>
+ * @author Pierrick Brihaye <pierrick.brihaye@free.fr>
+ */
+abstract class BrokerPools {
+
+    private static final Logger LOG = LogManager.getLogger(BrokerPools.class);
+
+    private static final ReadWriteLock instancesLock = new ReentrantReadWriteLock();
+    @GuardedBy("instancesLock") private static final Map<String, BrokerPool> instances = new TreeMap<>();
+
+    /**
+     * The name of a default database instance
+     */
+    public static String DEFAULT_INSTANCE_NAME = "exist";
+
+    // register a shutdown hook
+    static {
+        try {
+            Runtime.getRuntime().addShutdownHook(new Thread("BrokerPools-ShutdownHook") {
```

Comment:
@shabanovd That is still used to control the Jetty shutdown hook (`JettyStart`), however that could most likely also be cleaned up to just unconditionally add the shutdown hook.